lib/ansible/modules/packaging/language/easy_install.py

```diff
@@ -94,17 +94,21 @@
 import os
 import os.path
 import tempfile
-
 from ansible.module_utils.basic import AnsibleModule
 
 
-def _is_package_installed(module, name, easy_install, executable_arguments):
-    executable_arguments = executable_arguments + ['--dry-run']
+def install_package(module, name, easy_install, executable_arguments):
     cmd = '%s %s %s' % (easy_install, ' '.join(executable_arguments), name)
-    rc, status_stdout, status_stderr = module.run_command(cmd)
+    rc, out, err = module.run_command(cmd)
+    return rc, out, err
+
+
+def _is_package_installed(module, name, easy_install, executable_arguments):
+    executable_arguments += ['--dry-run']
```

Comment:
One little bug here:

executable_arguments is a list which is a mutable container.  This adds ['--dry-run'] to that container.  In python, mutable containers are passed by reference.  So in the caller, after _is_package_installed() is called, executable_arguments should include '--dry-run'.  That's obviously undesirable.

The previous code rebound the identifier executable_arguments to a new list which contained a copy of the parameter executable_arguments with the additional element '--dry-run'.

It's probably best to make it more obvious that this code should operate on a copy of executable_arguments.  Perhaps:

``` python
def _is_package_installed(module, name, easy_install, executable_arguments):
    # Copy and add to the arguments
    executable_arguments = executable_arguments[:]
    executable_arguments.append('--dry-run')
    rc, out, err = install_package(module, name, easy_install, executable_arguments)
    if rc:
        module.fail_json(msg=err)
    return 'Downloading' not in out
```